### PR TITLE
feat(Drawer): add closeIcon prop

### LINKED ISSUE #19153: Add prop to change close icon for Drawer
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
Our design has a custom icon for our mobile sidebar's close button. It would be great if there were a prop to change the close icon for the Drawer component.

### What does the proposed API look like?
A prop for rendering the icon. E.g:

```jsx
<Drawer icon={<SomeIcon />} />
```

### PR DESCRIPTION
**Pull request by kyle-copeland**
### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
close #19153

### 💡 Background and solution

1. Users are currently unable to customize the Drawer close icon
2. Component in action
Annotation 2019-10-17 222934

3. New API
```jsx
<Drawer closeIcon={<Icon type="stop" />}>
  Content
</Drawer>
```

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    add closeIcon prop to Drawer component       |
| 🇨🇳 Chinese |      添加关闭图标支持关闭当前 Drawer    |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/drawer/demo/custom-close-icon.md](https://github.com/kyle-copeland/ant-design/blob/feature-drawer-icon/components/drawer/demo/custom-close-icon.md)
[View rendered components/drawer/index.en-US.md](https://github.com/kyle-copeland/ant-design/blob/feature-drawer-icon/components/drawer/index.en-US.md)
[View rendered components/drawer/index.zh-CN.md](https://github.com/kyle-copeland/ant-design/blob/feature-drawer-icon/components/drawer/index.zh-CN.md)

### CODE REVIEW

**Comment by kyle-copeland** [comment]
Needs translations <3

**Comment by kyle-copeland** [comment]
Would anyone be able to help me with the failing tests? The tests seem unrelated to the component I updated. Running `npm test` locally does not seem to result in test failures.

**Comment by yoyo837** [comment]
reinstall your dependencies and `npm test -- -u`

**Comment by kyle-copeland** [comment]
@yoyo837 It looks like some of the tests were failing due to some tests that utilize the `role=spinbutton`  but the build still fails on CircleCI after updating those tests.

Here are my results locally when running `npm test -- -u` :
Capture

Here are the results from CircleCI:
Capture

**Comment by yoyo837** [comment]
Please try to remove the local `node_modules` and reinstall, and run `npm test -- -u` again.

**Comment by kyle-copeland** [comment]
Thanks for the reply @yoyo837. I ran `rm -rf node_modules` and `npm install` `npm test -- -u` and I got the same results as the previous screenshot. I noticed that the tests did past on CircleCI for the `test_*_15` builds. Do you know how those are different from the `test_*` builds?

Also is there a process for getting the documentation translated =]?

**Comment by afc163** [comment]
`/rebase` cannot works on forked branch.

**Comment by yoyo837** [comment]
It seems that I misunderstood this directive.   :)

**Comment by kyle-copeland** [comment]
Thank you for adding the translations =]. I will rebase tonight

**Comment by kyle-copeland** [comment]
@yoyo837  I am still experiencing issues with my tests sorry. Do you have any thoughts on why this might be? I wonder if it has to do with me using wsl on Windows?

I've tried using both node 10 and 12 running `rm -rf node_modules`, `npm install`, `npm test -- -u` and it doesn't update the snapshots of the failing tests in circleci. The tests all pass on my machine.

File: components/drawer/index.tsx [hunk 1]
```diff
@@ -46,6 +46,7 @@ export interface DrawerProps {
   className?: string;
   handler?: React.ReactNode;
   keyboard?: boolean;
+  closeIcon?: React.ReactElement<any>;
```
Comment (dengfuping) [inline components/drawer/index.tsx]:
```diff
- closeIcon?: React.ReactElement<any>;
+ closeIcon?: React.ReactNode;
```

File: components/drawer/index.en-US.md [hunk 2]
```diff
@@ -37,6 +37,7 @@ A Drawer is a panel that is typically overlaid on top of a page and slides in fr
 | zIndex | The `z-index` of the Drawer. | Number | 1000 | 3.7.0 |
 | placement | The placement of the Drawer. | 'top' \| 'right' \| 'bottom' \| 'left' | 'right' | 3.7.0 |
 | onClose | Specify a callback that will be called when a user clicks mask, close button or Cancel button. | function(e) | - | 3.7.0 |
+| closeIcon | Custom close icon | React.ReactElement | - |  |
```
Comment (dengfuping) [inline components/drawer/index.en-US.md]: - `React.ReactElement` => `React.ReactNode`

File: components/drawer/index.zh-CN.md [hunk 3]
```diff
@@ -36,6 +36,7 @@ title: Drawer
 | zIndex | 设置 Drawer 的 `z-index` | Number | 1000 | 3.7.0 |
 | placement | 抽屉的方向 | 'top' \| 'right' \| 'bottom' \| 'left' | 'right' | 3.7.0 |
 | onClose | 点击遮罩层或右上角叉或取消按钮的回调 | function(e) | 无 | 3.7.0 |
+| closeIcon | 自定义关闭icon | React.ReactElement | - |  |
```
Comment (dengfuping) [inline components/drawer/index.zh-CN.md]: - 中英文之间要用逗号分隔: `自定义关闭icon` => `自定义关闭 icon`
- `React.ReactElement` => `React.ReactNode`

File: components/drawer/demo/custom-close-icon.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
+---
+order: 6
+title:
+  zh-CN: 自定义Drawer关闭icon
```
Comment (dengfuping) [inline components/drawer/demo/custom-close-icon.md]: - `自定义Drawer关闭icon` => `自定义关闭图标`

File: components/drawer/demo/custom-close-icon.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  en-US: Custom close icon
+---
+
+## zh-CN
+
+自定义Drawer关闭icon
```
Comment (dengfuping) [inline components/drawer/demo/custom-close-icon.md]: - `自定义Drawer关闭icon` => `自定义关闭图标`

File: components/drawer/demo/custom-close-icon.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+## en-US
+
+Customize close icon of drawer
```
Comment (dengfuping) [inline components/drawer/demo/custom-close-icon.md]: - `Customize close icon of drawer` => `Customize close icon`

File: components/drawer/index.tsx (hunk 1 above)
Comment (kyle-copeland) [reply components/drawer/index.tsx]: Thanks for the feedback @dengfuping =]

**Pull request closed without merging**